tests/queries/0_stateless/01104_distributed_numbers_test.sql

```diff
@@ -1,9 +1,25 @@
 -- Tags: distributed
 
+SELECT *
+FROM
+(
+    SELECT *
+    FROM system.numbers
+    WHERE number = 100
+    UNION ALL
+    SELECT *
+    FROM system.numbers
+    WHERE number = 100
```

Comment:
Apparently, after https://github.com/ClickHouse/ClickHouse/pull/50909, this does not check what it was intended to.